frame/transaction-payment/src/lib.rs

```diff
@@ -445,6 +445,39 @@ where
 		}
 	}
 
+	/// Query runtime dispatch info of a given `Call`.
+	pub fn query_call_info<Call: GetDispatchInfo>(
+		call: Call,
+		len: u32,
+	) -> RuntimeDispatchInfo<BalanceOf<T>>
+	where
+		T::Call: Dispatchable<Info = DispatchInfo>,
```

Comment:
And here you also use the `T::Call`. You should only use this and skip the custom generic `Call` as @kianenigma already said.